File: UC04-Logica_Programacao/Java_Python/investimento.py
# Programa para calcular investimento
# para um determinado valor inicial
#quanto tempo demora para atingir o valor desejado

#variaveis de controle
periodo = 12
continua = 1

# entrada dos valores de investimento
print("Digite valores do investimento")
valorInicial=float(input("Valor inicial de investimento : "))
valorDeposito=float(input("Valor deposito mensal : "))
taxaJuros=float(input("Taxa de juros mensal :"))

# cacula e apresenta valores 
while continua == 1:
    for n in range(periodo): 
     valorInicial = (valorInicial*taxaJuros) + valorDeposito + valorInicial
     print("Valor mes : ", n+1 , " ", round(valorInicial,2))
    
    # pergunta sobtre continuidade da simulação 
    # apos termino do laco 
    print("______________________________________")
    continua = int(input("Simular mais um ano ? Se sim digite 1 : "))
    if continua != 1 :
     print("\n Simulação encerrada")
    

File: UC04-Logica_Programacao/Java_Python/Investimento.java
import java.util.Scanner;

public class Investimento {
// Programa para calcular investimento
// para um determinado valor inicial
// quanto tempo demora para atingir o valor desejado

    public static void main(String[] args) {
        // variaveis de controle
        int periodo = 12, continua = 1;

        // Variaveis de investimento 
        float valorDesejado, valorInicial, valorDeposito, taxaJuros;

        // ler do teclado
        Scanner ler = new Scanner(System.in);
        
        // entrar com valores
        System.out.print("\n Valor Final desejado : ");
        valorDesejado = ler.nextFloat();
        System.out.print("\n Valor inicial de investimento : ");
        valorInicial = ler.nextFloat();
        System.out.print("\n Valor de depósito mensal : ");
        valorDeposito = ler.nextFloat();
        System.out.print("\n Taxa de juros mensais : ");
        taxaJuros = ler.nextFloat();
       
        
        // calcular valor para cada 12 meses
        while (continua == 1) {
            System.out.println(" ");
            for ( int i = 1; i <= periodo; i++) {
                valorInicial = (valorInicial*taxaJuros) + valorDeposito + valorInicial;
                System.out.printf("Valor do mês "+ i + " : %.2f %n", valorInicial);
            }
            System.out.println("---------------------------");
            // confirmar se quer simular mais 1 ano 
            if ( valorInicial < valorDesejado) {
                System.out.print("\n Valor desejado ainda não alcançado");
                System.out.print("\n Simular mais 1 ano ? Se sim digite 1  : ");
                continua = ler.nextInt();
            }
            else{ 
                System.out.println("\n Valor desejado alcançado");
                continua = 0 ;
            }
           
            if (continua == 0 ) { System.out.println("\n Simulação encerrada \n");}
        }   
        
    }
}